### DISCUSSION TITLE
Custom controller does not receive events when claims and XRs are updated

### DISCUSSION THREAD

**Discussion by jsanda** [crossplane/crossplane#5412] [General]
I am using crossplane 1.15. I have developed several XRDs and integrating them with a custom controller that I am developing with the controller-runtime library. I created client structs for my XRDs by hand and am using them in controller. For the most part things seem to be working well; however, I noticed recently that reconciliation isn't getting triggered when my claims and XRs get updated.

I set up my watches as follows:

```go
func (r *MyReconciler) SetupWithManager(mgr ctrl.Manager) error {
	return ctrl.NewControllerManagedBy(mgr).
		For(&v1alpha1.MyCustomResource{}).
		Owns(&v1alpha1.MyXRD{}).
		Complete(r)
}
```

I suspect that I need to configure my watch differently, but I am not sure how. Any insights would be greatly appreciated!

**Comment by Joro97** [comment]
@jbw976 can you please help answer? This seems like it could be a potential bug. One would expect that updates to the XRs would trigger a reconciliation.

**Reply by dalemcgavin** [reply]
Can I see a snippet of you creating  the XRD programmatically, I think Owns requires you add Owner references to the XRD upon creation.


If you're custom resource is namespace scoped you'll run into issues because XRD is cluster scoped and therefore can't own an XRD. 

If that's the case then claims are namespace scoped and you should be able create and own them no problem

Please tell me if I'm wrong because I'm trying to do a similar thing
